apps/web/server/routers/viewer.tsx

```diff
@@ -940,6 +942,244 @@ const loggedInViewerRouter = createProtectedRouter()
 
       return locationOptions;
     },
+  })
+  .mutation("deleteCredential", {
+    input: z.object({
+      id: z.number(),
+    }),
+    async resolve({ input, ctx }) {
+      const { id } = input;
+
+      const credential = await prisma.credential.findFirst({
+        where: {
+          id: id,
+        },
+        include: {
+          app: {
+            select: {
+              slug: true,
+              categories: true,
+            },
+          },
+        },
+      });
+
+      if (!credential) {
+        throw new TRPCError({ code: "INTERNAL_SERVER_ERROR" });
+      }
+
+      if (credential.userId !== ctx.user.id) {
+        throw new TRPCError({ code: "UNAUTHORIZED" });
+      }
+
+      const eventTypes = await prisma.eventType.findMany({
+        where: {
+          users: {
+            some: {
+              id: ctx.user.id,
+            },
+          },
+        },
+        select: {
+          id: true,
+          locations: true,
+          destinationCalendar: true,
+          price: true,
+        },
+      });
+
+      // If it's a video, replace the location with Cal video
+      if (credential.app?.categories.includes(AppCategories.video)) {
+        // Find the user's event types
+
+        // Look for integration name from app slug
+        const integrationQuery =
+          credential.app?.slug === "msteams" ? "office365_video" : credential.app?.slug.split("-")[0];
+
+        // Check if the event type uses the deleted integration
+        for (const eventType of eventTypes) {
+          if (isPrismaArray(eventType.locations)) {
+            // To avoid type errors, need to stringify and parse JSON to use array methods
+            const locationsSchema = z.array(z.object({ type: z.string() }));
```

Comment:
@zomars is this what you mean by using zod?

Reply:
Yes @joeauyeung also since we're already using the schema there's no need for `isPrismaArray` anymore